src/qibocal/fitting/classifier/plots.py

```diff
@@ -59,43 +59,50 @@ def plot_conf_matr(y_test, base_dir: pathlib.Path, classifiers=None):
     )
 
 
-def plot_roc_curves(y_test, base_dir: pathlib.Path, classifiers=None):
+def plot_roc_curves(x_test, y_test, base_dir: pathlib.Path, models, names):
     _figure = plt.figure(figsize=(30, 5))
     fprs = []
     tprs = []
-    names = []
-
-    if classifiers is None:
-        classifiers = [i.value for i in Classifiers]
+    # names = []
 
-    len_list = len(classifiers)
+    # if classifiers is None:
+    #     classifiers = [i.value for i in Classifiers]
 
-    for count, model in enumerate(classifiers):
-        classifier = run.Classifier(model, base_dir)
-        names.append(classifier.name)
-        y_pred = np.load(base_dir / classifier.name / run.PREDFILE)
+    # len_list = len(classifiers)
+    len_list = len(models)
+    for count, model in enumerate(models):
+        # classifier = run.Classifier(model, base_dir)
+        # names.append(names[count])
 
         ax = plt.subplot(1, len_list, count + 1)
         plt.subplot(1, len_list, count + 1)
 
+        y_pred = np.load(base_dir / names[count] / run.PREDFILE)
         fpr, tpr, _ = roc_curve(
             y_test, y_pred
         )  # TODO: add dictionary and Ramiro's method and close windows with plt.close('all')
         fprs.append(fpr)
         tprs.append(tpr)
-        # roc_auc = auc(fpr, tpr)
-        # _roc_display = RocCurveDisplay(fpr=fpr, tpr=tpr, roc_auc=roc_auc, ax=ax,color="darkorange")
-        RocCurveDisplay.from_predictions(
-            y_test,
-            y_pred,
-            ax=ax,
-            color="darkorange",
-        )
+
+        print(model)
+        try:
+            RocCurveDisplay.from_estimator(
+                model, x_test, y_test, ax=ax, color="darkorange"
+            )
```

Comment:
What if you try to do:
```suggestion
			y_pred = model.decision_function(x_test)
            RocCurveDisplay.from_predictions(
                y_test, y_pred, ax=ax, color="darkorange"
            )
```

like in here?
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.RocCurveDisplay.html#sklearn.metrics.RocCurveDisplay.from_predictions

Reply:
I tried, but the neural network and `qubit fit` have not the `decision_function` method.

Reply:
Yes, I obtained the same results, but it works only with the scikit-learn classifiers.

Reply:
That is fine and expected, but then how did you obtain `y_pred` before for the scikit ones?

Reply:
I used `predict` method. I understand your point, I will change it soon.

Reply:
At this point, it would also be interesting to understand the difference between `.predict()` and `.decision_function()`. I would have also expected to have the same effect for a classifier, but maybe `.predict()` is returning something else...

Reply:
It looks like they're methods of different objects:
- `.predict()` is a method of a calibrated classifier (that wraps an estimator)
- `.decision_function()` of an estimator

Why they give back different results, I still don't know
https://scikit-learn.org/stable//modules/generated/sklearn.calibration.CalibratedClassifierCV.html

Reply:
> Predict the target of new samples.
> The predicted class is the class that has the highest probability, and can thus be different from the prediction of the uncalibrated classifier.

https://scikit-learn.org/stable//modules/generated/sklearn.calibration.CalibratedClassifierCV.html#sklearn.calibration.CalibratedClassifierCV.predict

Not sure if we need this calibration, and how are we doing it, but the plots definitely seemed weird with `.predict()`...